### PR TITLE
ORV2-2061: Route to different auth providers from home page based on route hint

### PR DESCRIPTION
**Pull request by erikataot**
# Description

- Updated the static `home.html` page to remove the hardcoded links to prod
- Added a `idpHint` search parameter to the query string to signal which auth to use; `idir', `bceidboth` or `none`
- Route from our default route based on that

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1296-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1296-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1296-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1296-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/src/common/components/searchparamrouter/SearchParamRouter.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { useEffect } from "react";
+import { useLocation, useNavigate } from "react-router-dom";
+import { IDPS } from "../../types/idp";
+import { useAuth } from "react-oidc-context";
+import { InitialLandingPage } from "../../../features/homePage/InitialLandingPage";
+
+const SearchParamRouter = () => {
```
Comment (erikataot) [inline frontend/src/common/components/searchparamrouter/SearchParamRouter.tsx]: maybe call it `HomeRouter`? I dunno - `SearchParamRouter` is technically accurate...

Comment (krishnan-aot) [reply frontend/src/common/components/searchparamrouter/SearchParamRouter.tsx]: `IDPRedirect` ? I'd rather call it that based on what it does.

Comment (erikataot) [reply frontend/src/common/components/searchparamrouter/SearchParamRouter.tsx]: sounds good to me 👍

File: frontend/src/common/components/idpredirect/IDPRedirect.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                    extraQueryParams: { kc_idp_hint: IDPS.IDIR },
+                });
+            break;
+            case IDPS.BCEID:
+                signinRedirect({
+                    extraQueryParams: { kc_idp_hint: IDPS.BCEID },
+                });
+            break;
+            default:
+            break;
```
Comment (krishnan-aot) [inline frontend/src/common/components/idpredirect/IDPRedirect.tsx]:
```suggestion
        switch(paramValue) {
            case IDPS.IDIR:
            case IDPS.BCEID:
                signinRedirect({
                    extraQueryParams: { kc_idp_hint: paramValue },
                });
            break;
            
            default:
            break;
```

**Review by krishnan-aot — APPROVED** [review]

**Review by zgong-gov — APPROVED** [review]

**Pull request merged**